File: backend/src/main/java/com/kelifang/notice/entity/Notice.java
package com.kelifang.notice.entity;

import lombok.Data;

import java.time.LocalDateTime;

/**
 * 站内通知。
 *
 * **刻意不落库**：demo 只要"发布方发出去、学生和家长能看见"，
 * 数据活在 {@code NoticeService} 的进程内列表里，重启即清空 —— 和 H2 内存库一个待遇。
 * 所以这里没有 @TableName，也没有对应的 Mapper。
 */
@Data
public class Notice {

    private Long id;

    private String title;

    private String content;

    /** 面向对象：ALL=全体（学生 + 家长） / STUDENT=仅学生 / PARENT=仅家长 */
    private String audience;

    /** 发布者。后端从登录态填，前端只传 title / content / audience。 */
    private Long publisherId;
    private String publisherName;
    private String publisherRole;

    private LocalDateTime createdAt;
}

File: frontend/src/api/notice.ts
import { del, get, post } from './request'

/** 站内通知。后端存在内存里，重启即清空。 */
export interface Notice {
  id?: number
  title: string
  content?: string
  /** ALL=全体（学生 + 家长） / STUDENT=仅学生 / PARENT=仅家长 */
  audience: string
  /** 发布者，由后端按登录态填，前端不传 */
  publisherName?: string
  publisherRole?: string
  createdAt?: string
}

export const AUDIENCE_OPTIONS = [
  { label: '全体', value: 'ALL' },
  { label: '仅学生', value: 'STUDENT' },
  { label: '仅家长', value: 'PARENT' }
]

export const AUDIENCE_LABEL: Record<string, string> = {
  ALL: '全体',
  STUDENT: '仅学生',
  PARENT: '仅家长'
}

export const noticeApi = {
  /** 发布方拿到全部通知，学生 / 家长只拿到发给自己的 */
  list: () => get<Notice[]>('/notice/list'),
  publish: (data: Notice) => post<Notice>('/notice', data),
  remove: (id: number) => del<void>(`/notice/${id}`)
}
